### ISSUE TITLE
Consider allowing more flexible tagging of resources

### ISSUE THREAD

**Issue by jen20** [pulumi/pulumi-awsx#306]
`awsx.ec2.Vpc` adds `Name` and `type` tags to several resources. There doesn't seem to be a way to prevent this behaviour, or modify it to prevent triggering my tagging OCD.

- The `Name` tag for AWS resources is displayed in the console, and it is common practice to use a natural language name rather than a composed key of some kind - right now we produce names such as `ci-vpc-public-0` rather than `CI Infrastructure Public Subnet 1` (indexing from 1 is also very common for things like subnets).

- The `type` tag is fine, except for having an inconsistent case.

Both of these are clearly situation dependent, so I think it would be better to generally allow more flexibility in tags we automatically apply to things to make the component widely usable.

**Comment by joeduffy** [comment]
Another example -- this taken from `ApplicationLoadBalancer` -- by default, if I name the ALB `web`, I get the following: `name` of `"ca84d134-556e886"` and `tags` of `{ "Name": "web" }`.

Per https://github.com/pulumi/pulumi-awsx/issues/296, this name isn't terribly useful. It's good this can be overridden, but I'd honestly prefer if we just use a good name. I understand there are length restrictions and you want to be able to go side by side. AFAIK, that's 32, which seems long enough for most useful names. We could just add maybe 3 characters of randomness to the end, plus the `-`, giving the user 28 characters to play with, and we could just throw if they exceeded this. E.g., this example would have yielded `web-ca8`, which I'd have been happy with (and having the ability to suppress auto-naming lets me eliminate the `-ca8`, yielding just `web`, if I truly desire).

Depending on how you created your ALB, `tags` may not be easily overridable (e.g., `ApplicationListener` doesn't offer a `tags` property to override -- even though it internally creates an ALB object which may need to be tagged).

And even then, it appears that the code unconditionally overwrites the `Name` tag, so there's actually no way for me to override these, even if I specify `tags: { "Name": "My Descriptive Tag "}`.

Repro:

```
import * as awsx from "@pulumi/awsx";

// Create a load balancer on port 80 and spin up two instances of Nginx.
const lb = new awsx.elasticloadbalancingv2.ApplicationLoadBalancer("web", {
    tags: { "Name": "My WWW Load Balancer" },
});
const listener = lb.createListener("web-listener", { port: 80 });
const nginx = new awsx.ecs.FargateService("nginx", {
    taskDefinitionArgs: {
        containers: {
            nginx: {
                image: "nginx",
                memory: 128,
                portMappings: [ listener ],
            },
        },
    },
    desiredCount: 2,
});

export const url = listener.endpoint.hostname;
export const tags = lb.loadBalancer.tags;
```

Notice the resulting `tags` overwrites my values specified above.

So, TL;DR, I do agree with @jen20 there is some more flexibility needed in this area.

**Comment by lukehoban** [comment]
See also https://github.com/pulumi/pulumi-awsx/issues/161 and #322.
